minion-certificate-verifier/main/src/main/java/org/opennms/horizon/minioncertverifier/controller/CertificateController.java

```diff
@@ -68,12 +76,30 @@ public ResponseEntity<Void> validateDebug(HttpServletRequest request) throws Exc
             logger.info("\t{}: {}", header, request.getHeader(header));
         }
 
-        String subjectDn = request.getHeader(headerName);
-        return validate(subjectDn);
+        String cert = request.getHeader("ssl-client-cert");
+        return validate(cert);
     }
 
     @GetMapping
-    public ResponseEntity<Void> validate(@RequestHeader("ssl-client-subject-dn") String clientSubjectDn) throws Exception {
+    public ResponseEntity<Void> validate(@RequestHeader("ssl-client-cert") String certificatePem) {
+        var span = Span.current();
+        CertificateParser parser;
+        try {
+            parser = new CertificateParser(certificatePem);
```

Comment:
If we can successfully parse the serial number, always add it as a span attribute. Then you can always search traces based on serial number and it will be available for debugging.

```suggestion
            parser = new CertificateParser(certificatePem);
            span.setAttribute("serial-number", parser.getSerialNumber());
```

Reply:
done